gcc/rust/checks/errors/borrowck/rust-bir-dump.cc

```diff
@@ -316,26 +300,68 @@ Dump::visit (CallExpr &expr)
 }
 
 void
-Dump::visit (Operator<1> &expr)
+Dump::visit (const Operator<1> &expr)
 {
   stream << "Operator(";
   visit_move_place (expr.get_operand<0> ());
   stream << ")";
 }
 
 void
-Dump::visit (Operator<2> &expr)
+Dump::visit (const Operator<2> &expr)
 {
   stream << "Operator(";
   visit_move_place (expr.get_operand<0> ());
   stream << ", ";
   visit_move_place (expr.get_operand<1> ());
   stream << ")";
 }
+
 void
-Dump::visit (Assignment &expr)
+Dump::visit (const Assignment &expr)
 {
-  visit_move_place (expr.get_rhs ());
+  if (func.place_db[expr.get_rhs ()].is_rvalue ())
+    {
+      visit_move_place (expr.get_rhs ());
+    }
+  else
+    {
+      visit_place (expr.get_rhs ());
+    }
```

Comment:
Nit
```suggestion
  if (func.place_db[expr.get_rhs ()].is_rvalue ())
      visit_move_place (expr.get_rhs ());
  else
      visit_place (expr.get_rhs ());
```

Reply:
Is this some kind of guideline? I really dislike if/else without braces.

Reply:
yes, it's from the GNU coding style guidelines. we've had comments about this in the past from GCC maintainers

Reply:
Eh, OK.

Reply:
Really I hate it too, makes it too easy to delete a line and get unexpected behavior on big codebase with a lot of warnings.